centreon/www/install/php/Update-25.01.0.php

```diff
@@ -0,0 +1,128 @@
+require_once __DIR__ . '/../../class/centreonLog.class.php';
+
+$centreonLog = CentreonLog::create();
+
+// error specific content
+$versionOfTheUpgrade = 'UPGRADE - 25.01.0: ';
+$errorMessage = '';
+
+$createUserProfileTable = function (CentreonDB $pearDB) use (&$errorMessage): void
+{
+    $errorMessage = 'Unable to create user_profile table';
+    $pearDB->executeQuery(
+        <<<'SQL'
+                CREATE TABLE IF NOT EXISTS `user_profile` (
+                  `id` INT UNSIGNED NOT NULL AUTO_INCREMENT,
+                  `contact_id` INT(11) NOT NULL,
+                  PRIMARY KEY (`id`),
+                  UNIQUE KEY `unique_user_profile` (`id`, `contact_id`),
+                  CONSTRAINT `fk_user_profile_contact_id`
+                    FOREIGN KEY (`contact_id`)
+                    REFERENCES `contact` (`contact_id`) ON DELETE CASCADE
+                ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4
+            SQL
+    );
+};
+
+$createUserProfileFavoriteDashboardsTable = function (CentreonDB $pearDB) use (&$errorMessage): void
+{
+    $errorMessage = 'Unable to create user_profile_favorite_dashboards table';
+    $pearDB->executeQuery(
+        <<<'SQL'
+                CREATE TABLE IF NOT EXISTS `user_profile_favorite_dashboards` (
+                  `profile_id` INT UNSIGNED NOT NULL,
+                  `dashboard_id` INT UNSIGNED NOT NULL,
+                  CONSTRAINT `fk_user_profile_favorite_dashboards_profile_id`
+                    FOREIGN KEY (`profile_id`)
+                    REFERENCES `user_profile` (`id`) ON DELETE CASCADE,
+                  CONSTRAINT `fk_user_profile_favorite_dashboards_dashboard_id`
+                    FOREIGN KEY (`dashboard_id`)
+                    REFERENCES `dashboard` (`id`) ON DELETE CASCADE
+                ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4
+            SQL
+    );
+};
+
+$createDefaultProfileForUsers = function (CentreonDB $pearDB) use (&$errorMessage): void
+{
+    $errorMessage = 'Failed to retrieve IDs of configured contacts on the platform';
+    $statement = $pearDB->executeQuery(
+        <<<'SQL'
+                SELECT contact_id FROM contact
+            SQL
+    );
+
+    $contactIds = $statement->fetchAll(PDO::FETCH_COLUMN);
+
+    $errorMessage = 'Unable to create default user profile for configured contacts of the platform';
+
+    foreach ($contactIds as $index => $contactId) {
+        $bindValues["contact_id_{$index}"] = $contactId;
+        $subQueries[] = "(:contact_id_{$index})";
+    }
+
+    $bindSubQuery = implode(', ', $subQueries);
+
+    $userProfileStatement = $pearDB->prepareQuery(
+        <<<SQL
+                INSERT INTO `user_profile` (contact_id) VALUES {$bindSubQuery}
+            SQL
+    );
+
+    foreach ($bindValues as $bindKey => $bindValue) {
+        $statement->bindValue($bindKey, $bindValue, PDO::PARAM_INT);
+    }
+
+    $pearDB->executePreparedQuery($userProfileStatement, array_keys($bindValues));
```

Comment:
You need to check if the contact ids are already in `user_profile` table (for idempotency reasons)

Reply:
Some changes will occur after some discussions the upgrade script will disappear :) I am putting back the PR in DRAFT state